scene/main/viewport.cpp

```diff
@@ -476,6 +478,19 @@ void Viewport::_notification(int p_what) {
 #endif // _3D_DISABLED
 		} break;
 
+		case NOTIFICATION_INTERNAL_PROCESS: {
+#ifndef _3D_DISABLED
+			if (use_xr && XRServer::get_singleton() != nullptr) {
+				Ref<XRInterface> xr_interface = XRServer::get_singleton()->get_primary_interface();
+				if (xr_interface.is_valid() && xr_interface->is_initialized()) {
+					Size2 xr_size = xr_interface->get_render_target_size();
+					// This will only actually resize if the the size has changed.
+					_set_size((Size2i)xr_size, Size2i(), Rect2i(), Transform2D(), true, xr_interface->get_view_count());
+				}
```

Comment:
Don't forget the else here to set the size to 0,0 if there is no valid xr_interface.

Reply:
Sure, I'll add all off these in my next update